test/DynamoCoreTests/CrashProtectionTests.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Dynamo.Models;
+using NUnit.Framework;
+
+namespace Dynamo.Tests
+{
+    [TestFixture]
```

Comment:
This file encodes all of the state transitions indicated above.  All of the tests succeed.

![image](https://cloud.githubusercontent.com/assets/916345/6835328/5ede8154-d310-11e4-8e5f-e86af997b3f9.png)